modules/openapi-generator/src/main/resources/typescript/README.mustache

```diff
@@ -28,3 +28,45 @@ _unPublished (not recommended):_
 
 ```
 npm install PATH_TO_GENERATED_PACKAGE --save
+
+### Usage
+
+Below code snippet shows exemplary usage of the configuration and the API based 
+on the typical `PetStore` example used for OpenAPI. 
+
+```
+import * as your_api from 'your_api_package'
+
+// Covers all auth methods included in your OpenAPI yaml definition
+const authConfig: your_api.AuthMethodsConfiguration = {
+    "api_key": "YOUR_API_KEY"
+}
+
+// Implements a simple middleware to modify requests before (`pre`) they are sent
+// and after (`post`) they have been received 
+class Test implements your_api.Middleware {
+    pre(context: your_api.RequestContext): Promise<your_api.RequestContext> {
+        // Modify context here and return
+        return Promise.resolve(context);
+    }
+
+    post(context: your_api.ResponseContext): Promise<your_api.ResponseContext> {
+        return Promise.resolve(context);
+    }
+
+}
+
+// Create configuration parameter object
+const configurationParameters = {
+    httpApi: new your_api.JQueryHttpLibrary(), // Can also be ignored - default is usually fine
+    baseServer: your_api.servers[0], // First server is default
+    authMethods: authConfig, // No auth is default
+    promiseMiddleware: [new Test()],
+}
+
+// Convert to actual configuration
+const config = your_api.createConfiguration(configurationParameters);
+
+// Use configuration with your_api
+const api = new your_api.PetApi(config);
```

Comment:
Would be great to also see how you would make a call off of this `api` instance.
It seems like we'd do something like `api.getUsers()`, but I'm getting tripped up by POST requests.

It seems like you'd do something like: 
```
// Seems like object arguments need to be namespaced, and that's based on the model name?
// If there's an option to not-namespace params I'd love to know it
api.createUser({createUserRequest: { email: 'email', password: 'password'}})
```

This seems to work for me if the SDK is making a call from one localhost port to another, but in my prod env it strips out the request payload. I imagine that's because I'm missing some kind of config option, and seeing how you do this would be super helpful information.

Reply:
I have added another couple lines with an example in the latest commit. Hope that explains it a bit more. There's also some examples in the `tests` folder under `samples/openapi3/client/petstore/typescript/` - hope this helps with debugging your problem.